sleuth_backend/views/views.py

```diff
@@ -122,67 +119,89 @@ def search(request):
+        state: (optional) any string to be returned in response
+
+    Example Usage:
+    http:// ... /api/get/?id=https://www.ubc.ca&return=links
+
+    Example Response Body:
+    {
+        "data": {
+            "type": <core name>,
+            "doc": { <document format depends on core> }
+        },
+        "request": { <data about your original request> }
+    }
     '''
-    kwargs = base_kwargs
-
-    if core == "genericPage":
-        fields = {
-            'id': 1,
-            'siteName': 10,
-            'name': 10,
-            'description': 5,
-            'content': 2
-        }
-        query = Query(query_str, fields=fields, proximity=5)
-        terms_query = Query(query_str, fields=fields, as_phrase=False)
-        query.select_or(terms_query)
-        kwargs['default_field'] = 'content'
-        kwargs['highlight_fields'] = 'content'
-
-    elif core == "courseItem":
-        fields = {
-            'id': 1,
-            'name': 9,
-            'description': 8,
-            'subjectData': 5,
-        }
-        query = Query(query_str, fields=fields)
-        kwargs['default_field'] = 'name'
-        kwargs['highlight_fields'] = 'description'
+    if request.method != 'GET':
+        return HttpResponse(status=405)
+
+    cores_to_search = build_core_request(request.GET.get('type', ''), SOLR.core_names())
+    doc_id = request.GET.get('id', '')
+    state = request.GET.get('state', '')
+    return_fields = build_return_fields(request.GET.get('return', ''))
+
+    kwargs = { 'return_fields': return_fields }
 
-    else:
-        query = Query(query_str)
+    if doc_id is '':
+        sleuth_error = SleuthError(ErrorTypes.INVALID_GETDOCUMENT_REQUEST)
+        return HttpResponse(sleuth_error.json(), status=400)
+
+    response = {
+        'data': {},
+        'request': {
+            'query': doc_id,
+            'types': cores_to_search,
+            'return_fields': return_fields.split(","),
+            'state': state
+        }
+    }
+    for core_to_search in cores_to_search:
+        try:
+            new_query, new_kwargs = build_getdocument_query(doc_id, kwargs)
+            query_response = SOLR.query(core_to_search, new_query, **new_kwargs)
+            if 'error' in query_response:
+                sleuth_error = SleuthError(
+                    ErrorTypes.SOLR_SEARCH_ERROR,
+                    message=query_response['error']['msg']+" on core "+core_to_search
+                )
+                return HttpResponse(sleuth_error.json(), status=query_response['error']['code'])
+            if query_response['response']['numFound'] != 0:
+                # Return result if a value is found
+                response['data'] = {
+                    'type': core_to_search,
+                    'doc': flatten_doc(query_response['response']['docs'][0], return_fields)
+                }
+                return JsonResponse(response)
```

Comment:
Why do you return before searching all the cores?

Reply:
`getdocument` should only ever return one document, since a URL is only parsed once into one core (so the same ID can't exist in both `genericPage` and `courseItem` for example)